third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -278,77 +143,41 @@ Value getScalarConstant(IRRewriter &rewriter, Location loc, Value expr) {
       return blockArg;
   }
 
-  return Value();
+  return {};
 }
 
 // Narrowing logic
 // For now we allow to narrow down to 32 bits only in the following case:
 // - `baseOffset` is 32-bits and `addOffset`(64-bits) is zero
+// TODO(max): is this correct?
```

Comment:
I think the comment is wrong (and should be rewritten), and the code is a bit convoluted and probably deserves at least a comment. The first time we use a non-uniform offset (and only the first time), we need to splat the zero offset and then add the non-uniform offset. I am catching the splat here .